src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -459,3 +466,61 @@ void function_pointer_restrictionst::write_to_file(
 
   function_pointer_restrictions_json.output(outFile);
 }
+function_pointer_restrictionst function_pointer_restrictionst::merge(
+  const function_pointer_restrictionst &other) const
+{
+  return function_pointer_restrictionst{
+    merge_function_pointer_restrictions(restrictions, other.restrictions)};
+}
+
+function_pointer_restrictionst get_function_pointer_by_name_restrictions(
+  const goto_modelt &goto_model,
+  const optionst &options)
```

Comment:
:bulb: I think it would be easier to see what is going on if you have this take the result of ` options.get_list_option(RESTRICT_FUNCTION_POINTER_BY_NAME_OPT)` to decouple the CLI from the parsing of the lsit